util/src/main/java/io/zeebe/util/sched/future/CompletableActorFuture.java

```diff
@@ -199,6 +201,15 @@ public void block(ActorTask onCompletion) {
     blockedTasks.add(onCompletion);
   }
 
+  @Override
+  public void onComplete(BiConsumer<V, Throwable> consumer) {
+    final ActorThread actorThread =
+        ActorControl.ensureCalledFromActorThread("CompletableActorFuture#onComplete");
+
+    final ActorControl actorControl = ActorControl.fromActorThread(actorThread);
```

Comment:
What about combining the two calls (getting the current actor thread + getting the control from the thread)?

```suggestion
    final ActorControl actorControl = ActorControl.current();
```

It's easier and exposes a bit less internals ;)